persistence/src/test/java/com/epam/esm/dao/TagDAOImplTest.java

```diff
@@ -0,0 +1,104 @@
+package com.epam.esm.dao;
+
+import com.epam.esm.config.TestConfig;
+import com.epam.esm.entity.Tag;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.util.List;
+import java.util.Optional;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {TestConfig.class})
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+@ActiveProfiles("dev")
+@Sql("classpath:/insert_data_certificate.sql")
+class TagDAOImplTest {
+
+    @Autowired
+    private TagDAO tagDAO;
+
+    @Test
+    void findAllValid() {
+        List<Tag> tags = tagDAO.findAll();
+        Assertions.assertEquals(2, tags.size());
+    }
+
+    @Test
+    void createNewValid() {
+        Tag tag = Tag.builder().name("tag").build();
+        Tag actual = tagDAO.create(tag);
+        Assertions.assertEquals(tag.getName(), actual.getName());
+    }
+
+    @Test
+    void createExistingValid() {
+        Tag tag = Tag.builder().name("IT").build();
+        Tag actual = tagDAO.create(tag);
+        Assertions.assertEquals(tag.getName(), actual.getName());
+    }
+
+    @Test
+    void findByIdValid() {
+        Long id = 1L;
+        Optional<Tag> actual = tagDAO.findById(id);
+        Assertions.assertEquals("IT", actual.get().getName());
+    }
+
+    @Test
+    void findByIdEmpty() {
```

Comment:
As I understood we try to find not existing tag. The name should demonstrate it

Reply:
done!